astropy/io/votable/tree.py

```diff
@@ -2459,11 +2460,12 @@ def ref(self, ref):
                 ref = None
             else:
                 self._fields = table.fields
+                self._all_fields = table.all_fields
                 self._params = table.params
                 self._groups = table.groups
                 self._links = table.links
         else:
-            del self._fields[:]
+            del self._all_fields[:]
```

Comment:
I think we have to `del` both to be consistent?
```suggestion
            del self._fields[:]
            del self._all_fields[:]
```

Reply:
indeed, thanks !